src/cmds/testing/block_dev/block_dev_test.c

```diff
@@ -146,7 +160,13 @@ static int block_dev_test(struct block_dev *bdev) {
 		return -1;
 	}
 
-	blocks = bdev->size / ((uint64_t) blk_sz);
+	total_blocks = bdev->size / ((uint64_t) blk_sz);
+
+	if (n_blocks)
+		blocks = min(s_block + n_blocks, total_blocks);
+	else
+		blocks = total_blocks;
+
```

Comment:
Also, I would suggest return immediately if the passed block count is 0.

Reply:
Oh..Is 'block count' compulsory argument? I thought it was an optional argument.

Reply:
You're right, it's an optional argument. But I mean how to deal with:

```
block_dev_test -n 0 hda
```

What's the expected behavior of this? As far as I can see it will start testing all block now.

Reply:
Or am I missing something?

Reply:
No, what you said is correct. I didn't think of this situation. I think we can mitigate this by doing check in the main function. I will update my commit.